src/utils.ts

```diff
@@ -67,3 +67,23 @@ export function cleanupRootStyles(rootId: string) {
     root.addEventListener('transitionend', onTransitionEnd);
   }
 }
+
+const useLayoutEffectSSR =
+  typeof window !== 'undefined' ? useLayoutEffect : useEffect;
+
+export function useWindowHeight() {
+  const [windowHeight, setWindowHeight] = useState(0);
+  useLayoutEffectSSR(() => {
+    setWindowHeight(window.innerHeight);
+  });
```

Comment:
This could be enhance with a window size listener so that the window height is in sync:

```suggestion
  useLayoutEffectSSR(() => {
    const updateHeight = () => setWindowHeight(window.innerHeight);
    window.addEventListener('resize', updateHeight);
    updateHeight();
    return () => window.removeEventListener('resize', updateHeight);
  }, []);
```